src/v/cluster/controller_backend.cc

```diff
@@ -487,12 +533,47 @@ ss::future<std::error_code> controller_backend::process_partition_update(
         co_return std::error_code(errc::waiting_for_recovery);
     }
 
+    /**
+     * Check if this NTP was moved from other shard on current broker, if so we
+     * have to wait for it to be shutdown. Partition creation will be dispatched
+     * to target partition manager from the core where the partition is
+     * currently present. On this core we will just wait for partition to be
+     * created.
+     */
+    if (has_node_local_replicas(_self, previous.replicas)) {
+        auto previous_shard = get_target_shard(_self, previous.replicas);
+        // ask previous controller for partition initial revision
+        auto initial_revision = co_await container().invoke_on(
+          *previous_shard, [ntp](controller_backend& remote) {
```

Comment:
is this lambda (temporary) an instance of the gcc+coroutine+temporary lifetime bug?